### DISCUSSION TITLE
v0.6.3

### DISCUSSION THREAD

**Discussion by takatost** [langgenius/dify#3522] [Releases]
# ✨ Spotlight on Dify v0.6 ✨

### Curious for more? Swing by the [v0.6.0 Release Notes](https://github.com/langgenius/dify/releases/tag/0.6.0) for the full scoop.

---

## What’s New in v0.6.3?

This update primarily focuses on enhancing system stability and improving user experience. Key updates include:

* feat: Add `JinaReader` as Tool by @Yeuoly in #3468
* feat: `Function calling` support for Google Gemini Pro by @Yeuoly in #3406
* feat: Add support for `embedding models` with AWS Bedrock Titan Model by @longzhihun in #3377
* feat: Support configurate openai compatible `stream tool call` by @Yeuoly in #3467
* feat: Refactor `tongyi` models, add `function calling & vision` support by @takatost in #3496
* feat: Add support for AWS Bedrock `Cohere embedding` by @kerlion in #3444
* feat: Cohere `rerank 3 model` added by @Yash-1511 in #3431
* feat: Add nvidia `codegemma 7b` support by @joshfeng in #3437
* feat: API tool support custom timeout by @Yeuoly in #3420
* feat: Integrated `SearXNG search` as built-in tool by @junytang in #3363
* feat: Add support for extracting `EPUB` files in RAG extractors by @vaayne in #3254
* feat: Add support for extracting `XLS` files in RAG extractors  by @ic-xu in #3321
* feat: Support `relyt` vector database by @klaus-xiong in #3367
* feat: Add workflow editor `shortcuts` (#3382) by @perzeuss in #3390
* feat: Show citation info in run history by @nite-knite in #3399
* feat: Optimize the efficiency of generating chatbot conversation name by @takatost in #3472

### Docker compose deployments:

1. Get the latest code from the main branch:

   ```Shell
   git checkout main
   git pull origin main
   ```

2. Go to the next step and update to the latest image:

   ```Shell
   cd docker
   docker compose up -d
   ```

### Source Code deployments:

1. Stop API server, Worker and Web frontend Server.

2. Get the latest code from the main branch:

   ```Shell
   git checkout main
   git pull origin main
   ```

3. Update Python dependencies:

   ```Shell
   cd api
   pip install -r requirements.txt
   ```

4. Then, let's run the migration script:

   ```Shell
   flask db upgrade
   ```

5. Finally, run API server, Worker and Web frontend Server again.

## What's Changed
* feat: Add support for embed file with AWS Bedrock Titan Model by @longzhihun in https://github.com/langgenius/dify/pull/3377
* fix: remove - in dataset retriever tool name by @Yeuoly in https://github.com/langgenius/dify/pull/3381
* feat:api Add support for extracting EPUB files in ExtractProcessor by @vaayne in https://github.com/langgenius/dify/pull/3254
* feat: show citation info in run history by @nite-knite in https://github.com/langgenius/dify/pull/3399
* Feat: Invitation link automatically completes domain name by @huangbaichao in https://github.com/langgenius/dify/pull/3393
* Integrated SearXNG search as built-in tool by @junytang in https://github.com/langgenius/dify/pull/3363
* fix: [azure_openai] Error: 'NoneType' object has no attribute 'content' by @sagarey in https://github.com/langgenius/dify/pull/3389
* Update providers preview by @guchenhe in https://github.com/langgenius/dify/pull/3403
* add xls file suport by @ic-xu in https://github.com/langgenius/dify/pull/3321
* Update README.md by @guchenhe in https://github.com/langgenius/dify/pull/3405
* Fix/workflow tool incorrect parameter configurations by @Yeuoly in https://github.com/langgenius/dify/pull/3402
* chore: replace all set interval by @iamjoel in https://github.com/langgenius/dify/pull/3411
* feat: Deprecate datetime.utcnow() in favor of datetime.now(timezone.utc).replace(tzinfo=None) for better timezone handling (#3408) by @pamaforce in https://github.com/langgenius/dify/pull/3416
* chore: remove Langchain tools import by @JohnJyong in https://github.com/langgenius/dify/pull/3407
* feat: gemini pro function call by @Yeuoly in https://github.com/langgenius/dify/pull/3406
* fix: shared text-generation stream by @zxhlyh in https://github.com/langgenius/dify/pull/3419
* fix/dataset-retriever-tool-parameter-redundancy by @Yeuoly in https://github.com/langgenius/dify/pull/3418
* Feat/api tool custom timeout by @Yeuoly in https://github.com/langgenius/dify/pull/3420
* fix: test env key missing or wrong by @takatost in https://github.com/langgenius/dify/pull/3430
* Doc/update readme by @crazywoola in https://github.com/langgenius/dify/pull/3433
* Update README_CN.md by @crazywoola in https://github.com/langgenius/dify/pull/3434
* Update README_CN.md by @crazywoola in https://github.com/langgenius/dify/pull/3435
* feat: add workflow editor shortcuts (#3382) by @perzeuss in https://github.com/langgenius/dify/pull/3390
* FEAT: cohere rerank 3 model added by @Yash-1511 in https://github.com/langgenius/dify/pull/3431
* chore: remove the COPY instruction in .devcontainer/Dockerfile by @BodhiHu in https://github.com/langgenius/dify/pull/3409
* fix typo: Changlog -> Changelog by @Selene29 in https://github.com/langgenius/dify/pull/3442
* fix: node shortcuts active in input fields by @perzeuss in https://github.com/langgenius/dify/pull/3438
* Add nvidia codegemma 7b by @joshfeng in https://github.com/langgenius/dify/pull/3437
* Update yaml and py file in Tavily Tool by @richards199999 in https://github.com/langgenius/dify/pull/3450
* feat: Added the mirror of Aliyun's Linux apk installation package and updated the deprecated taobao npm mirror address to npmmirror by @Kennytian in https://github.com/langgenius/dify/pull/3459
* Revert "Update yaml and py file in Tavily Tool" by @crazywoola in https://github.com/langgenius/dify/pull/3464
* feat: jina reader by @Yeuoly in https://github.com/langgenius/dify/pull/3468
* feat: support configurate openai compatible stream tool call by @Yeuoly in https://github.com/langgenius/dify/pull/3467
* feat: optimize the efficiency of generating chatbot conversation name by @takatost in https://github.com/langgenius/dify/pull/3472
* feat: remove langchain from output parsers by @takatost in https://github.com/langgenius/dify/pull/3473
* chore: separate Python dependencies for development by @bowenliang123 in https://github.com/langgenius/dify/pull/3198
* chore: add sandbox permission tooltip by @Yeuoly in https://github.com/langgenius/dify/pull/3477
* fix: prompt template issue by @YidaHu in https://github.com/langgenius/dify/pull/3449
* feat: support relyt vector database  by @klaus-xiong in https://github.com/langgenius/dify/pull/3367
* Update README.md by @guchenhe in https://github.com/langgenius/dify/pull/3478
* nvidia-label-update by @joshua20231026 in https://github.com/langgenius/dify/pull/3482
* fix: in conversation log click op button would cause close drawer by @iamjoel in https://github.com/langgenius/dify/pull/3483
* fix: workflow auto layout nodes offset & delete node shortcuts by @zxhlyh in https://github.com/langgenius/dify/pull/3484
* fix: workflow edge curvature by @zxhlyh in https://github.com/langgenius/dify/pull/3488
* fix: stringify object while exporting batch result to csv by @sinomoe in https://github.com/langgenius/dify/pull/3481
* question classifier prompt optimization by @JohnJyong in https://github.com/langgenius/dify/pull/3479
* feat: refactor tongyi models by @takatost in https://github.com/langgenius/dify/pull/3496
* fix: bump twilio to 9.0.4 skipping yanked versions by @bowenliang123 in https://github.com/langgenius/dify/pull/3500
* test: install ffmpeg for pytests by @bowenliang123 in https://github.com/langgenius/dify/pull/3499
* feat: support var auto rename in prompt editor by @iamjoel in https://github.com/langgenius/dify/pull/3510
* fix: add message caused problem after simple chat convert to workflow by @iamjoel in https://github.com/langgenius/dify/pull/3511
* fix: the object field is empty string in some openAI api compatible model by @liuzhenghua in https://github.com/langgenius/dify/pull/3506
* Add suuport for AWS  Bedrock Cohere embedding by @kerlion in https://github.com/langgenius/dify/pull/3444
* fix: add completion mode object check by @Yeuoly in https://github.com/langgenius/dify/pull/3515
* get config default for sandbox by @miendinh in https://github.com/langgenius/dify/pull/3508
* chore: improve reference variable picker user experience by @iamjoel in https://github.com/langgenius/dify/pull/3517
* fix: array[string] context in llm node invalid by @takatost in https://github.com/langgenius/dify/pull/3518
* version to 0.6.3 by @takatost in https://github.com/langgenius/dify/pull/3519

## New Contributors
* @longzhihun made their first contribution in https://github.com/langgenius/dify/pull/3377
* @vaayne made their first contribution in https://github.com/langgenius/dify/pull/3254
* @junytang made their first contribution in https://github.com/langgenius/dify/pull/3363
* @sagarey made their first contribution in https://github.com/langgenius/dify/pull/3389
* @ic-xu made their first contribution in https://github.com/langgenius/dify/pull/3321
* @pamaforce made their first contribution in https://github.com/langgenius/dify/pull/3416
* @BodhiHu made their first contribution in https://github.com/langgenius/dify/pull/3409
* @Selene29 made their first contribution in https://github.com/langgenius/dify/pull/3442
* @joshfeng made their first contribution in https://github.com/langgenius/dify/pull/3437
* @YidaHu made their first contribution in https://github.com/langgenius/dify/pull/3449
* @klaus-xiong made their first contribution in https://github.com/langgenius/dify/pull/3367
* @sinomoe made their first contribution in https://github.com/langgenius/dify/pull/3481
* @liuzhenghua made their first contribution in https://github.com/langgenius/dify/pull/3506
* @kerlion made their first contribution in https://github.com/langgenius/dify/pull/3444
* @miendinh made their first contribution in https://github.com/langgenius/dify/pull/3508

**Full Changelog**: https://github.com/langgenius/dify/compare/0.6.2...0.6.3

<hr /><em>This discussion was created from the release <a href='https://github.com/langgenius/dify/releases/tag/0.6.3'>v0.6.3</a>.</em>

**Comment by youkpan** [comment]
Nice job , by the way, Anyone want to try another style of AI workflow ? 
here is my website ,I build by myself: [https://ai.zyinfo.pro/wf](https://ai.zyinfo.pro/wf)
Wellcome to use ,haha 
![AI workflow](https://github.com/langgenius/dify/assets/2157423/dd725063-1a6e-4d38-970e-c3cfb01c8003)
